lib/cylc/cycling/integer.py

```diff
@@ -536,37 +564,77 @@ def get_point_from_expression(point_expr, context_point, is_required=False):
+
+    def test_exclusions_simple(self):
+        """Test the generation of points for integer sequences with exclusions.
+        """
+        sequence = IntegerSequence('R/P1!3', 1, 5)
+        output = []
+        point = sequence.get_start_point()
+        while point:
+            output.append(point)
+            point = sequence.get_next_point(point)
+        self.assertEqual([int(out) for out in output], [1, 2, 4, 5])
+
+    def test_exclusions_extensive(self):
+        """Test IntegerSequence methods for sequences with exclusions."""
+        point_0 = IntegerPoint(0)
+        point_1 = IntegerPoint(1)
+        point_2 = IntegerPoint(2)
+        point_3 = IntegerPoint(3)
+        point_4 = IntegerPoint(4)
+
+        sequence = IntegerSequence('R/P1!3', 1, 5)
+        self.assertFalse(sequence.is_on_sequence(point_3))
+        self.assertFalse(sequence.is_valid(point_3))
+        self.assertEqual(sequence.get_prev_point(point_3), point_2)
+        self.assertEqual(sequence.get_prev_point(point_4), point_2)
+        self.assertEqual(sequence.get_nearest_prev_point(point_3), point_2)
+        self.assertEqual(sequence.get_nearest_prev_point(point_3), point_2)
+        self.assertEqual(sequence.get_next_point(point_3), point_4)
+        self.assertEqual(sequence.get_next_point(point_2), point_4)
+        self.assertEqual(sequence.get_next_point_on_sequence(point_3), point_4)
+        self.assertEqual(sequence.get_next_point_on_sequence(point_2), point_4)
+
+        sequence = IntegerSequence('R/P1!1', 1, 5)
+        self.assertEqual(sequence.get_first_point(point_1), point_2)
+        self.assertEqual(sequence.get_first_point(point_0), point_2)
+        self.assertEqual(sequence.get_start_point(), point_2)
+
+        sequence = IntegerSequence('R/P1!5', 1, 5)
+        self.assertEqual(sequence.get_stop_point(), point_4)
+
+    def test_simple(self):
+        """Run some simple tests for integer cycling."""
+        sequence = IntegerSequence('R/1/P3', 1, 10)
+        start = sequence.p_start
+        stop = sequence.p_stop
+
+        # Test point generation forwards.
+        point = start
+        output = []
+        while point and stop and point <= stop:
+            output.append(point)
+            point = sequence.get_next_point(point)
+        self.assertEqual([int(out) for out in output], [1, 4, 7, 10])
+
+        # Test point generation backwards.
+        point = stop
+        output = []
+        while point and start and point >= start:
+            output.append(point)
+            point = sequence.get_prev_point(point)
+        self.assertEqual([int(out) for out in output], [10, 7, 4, 1])
+
+        # Test sequence comparison
+        sequence1 = IntegerSequence('R/1/P1', 1, 10)
+        sequence2 = IntegerSequence('R/1/P1', 1, 10)
+        self.assertEqual(sequence1, sequence2)
+        sequence2.set_offset(IntegerInterval('-P2'))
+        self.assertEqual(sequence1, sequence2)
```

Comment:
They should be equal as the offset is a multiple of the interval. I've added another test to cover the case where the offset is not a multiple.